docs/google.rst

```diff
@@ -136,14 +137,25 @@ google.batch.subnetwork                        Define the name of the subnetwork
 
 Process definition
 ------------------
-Processes can be defined as usual and by default the ``cpus`` and ``memory`` directives are used to find the cheapest machine
-type available at current location that fits the requested resources. If ``memory`` is not specified, 1GB of memory is allocated per cpu.
 
-As of version ``23.02.0-edge``, the process ``machineType`` directive can be a list of patterns separated by comma. The pattern can contain a ``*`` to match
-any number of characters and ``?`` to match any single character. Examples of valid patterns: ``c2-*``, ```m?-standard*``, ```n*``.
+Processes can be defined as usual. By default, the ``cpus`` and ``memory`` directives are used to find the cheapest machine
+type that is available at the current location and that fits the requested resources. If ``memory`` is not specified, 1GB of memory is allocated per CPU.
 
-Alternatively it can also be used to define a specific predefined Google Compute Platform `machine type <https://cloud.google.com/compute/docs/machine-types>`_
-or a custom machine type.
+The ``machineType`` process directive can be specified in any of the following ways:
+
+* A comma-separated list of patterns. The pattern can contain a ``*`` to match any number of characters
+  and ``?`` to match any single character. Examples of valid patterns: ``c2-*``, ``m?-standard*``, ``n*`` (requires  version ``23.02.0-edge`` or later).
+
+* A predefined Google Compute Platform `machine type <https://cloud.google.com/compute/docs/machine-types>`_ or `custom machine type <https://cloud.google.com/compute/docs/instances/creating-instance-with-custom-machine-type>`_.
+
+* An `instance template <https://cloud.google.com/compute/docs/instance-templates>`_, specified as ``template://<instance-template>``. Using an instance template
+  will overwrite the ``accelerator`` directive, as well as the following Google Batch options: ``cpuPlatform``, ``preemptible``, and ``spot``. To use an instance template
+  with GPUs, you must also set the ``google.batch.installGpuDrivers`` config option to ``true``. To use an instance template with Fusion, the instance template must include
+  a ``local-ssd`` disk named ``fusion`` with at least 375 GB. Requires version ``23.03.0.edge`` or later.
```

Comment:
@pditommaso I put a warning here in the docs about which options are overwritten by the instance template. I could add warnings for these settings in the code.

Directives like `cpus` and `memory` can be used with the instance template. In this case, if the machine type is bigger than the requested resources then Google Batch will try to pack multiple tasks on the same VM.

Reply:
I ran some tests and wasn't able to get Google Batch to pack tasks onto the same VM. Still, you can use `cpus` and `memory` with `machineType`. I might have been missing something else to make it work.